lightning/src/ln/outbound_payment.rs

```diff
@@ -1075,8 +1080,14 @@ impl OutboundPayments {
 				payment_id,
 				failed_paths_retry: if pending_amt_unsent != 0 {
 					if let Some(payment_params) = &route.payment_params {
+						let mut payment_params = payment_params.clone();
+						// We calculate the leftover fee budget we're allowed to spend by
+						// subtracting the used fee from the total fee budget.
+						payment_params.max_total_routing_fee_msat = payment_params
```

Comment:
Would be nice to get some test coverage on the `outbound_payment` changes, ok if we want to put it off though